src/cljs/main/broadfcui/main.cljs

```diff
@@ -204,6 +204,15 @@
   {:handle-hash-change
    (fn [{:keys [state]}]
      (let [window-hash (aget js/window "location" "hash")]
+       ;; if Terra redirects are globally enabled via config;
+       ;;   and the current hash has a path handler;
+       ;;   and that path handler has a Terra redirect,
+       ;; then redirect to Terra.
+       (when (config/terra-redirects-enabled)
+         (when-let [handler (nav/find-path-handler window-hash)]
+           (let [{:keys [make-props terra-redirect]} handler]
+             (when terra-redirect
+               (js-invoke (aget js/window "location") "replace" (str (config/terra-url) (terra-redirect (make-props))))))))
```

Comment:
I like the approach.

Looks like my cljs is rusty. How does `make-props` get its arguments? In this case, it needs `namespace` and `name` in order to construct the URL.

Reply:
`make-props` gets its arguments based on the regular expression capture matches, as defined by the `regex` in each path definition, e.g. https://github.com/broadinstitute/firecloud-ui/blob/develop/src/cljs/main/broadfcui/page/workspace/details.cljs#L190. The code to calculate the regex matches is at https://github.com/broadinstitute/firecloud-ui/blob/develop/src/cljs/main/broadfcui/nav.cljs#L34.

Reply:
Ah, thanks. What I was missing is that the `make-props` attribute returned from `find-path-handler` captures the arguments; it's not the same parameterized `make-props` from the call to `nav/defpath`.